discovery/marathon/marathon.go

```diff
@@ -98,6 +100,7 @@ func NewDiscovery(conf *config.MarathonSDConfig) (*Discovery, error) {
 		servers:         conf.Servers,
 		refreshInterval: time.Duration(conf.RefreshInterval),
 		appsClient:      fetchApps,
+		token:           os.Getenv(conf.TokenEnvName),
```

Comment:
We have to keep some consistency around handling secrets. There are a handful of secrets managed already this way. If it's the wrong way, we needed to change it everywhere, which would be a breaking change only possible in v2.0. We do not want to support many different ways of managing secrets as that would be confusing and hard to maintain.